main.cpp

```diff
@@ -5,20 +5,65 @@ int main_dialog::idd() const {
 	return IDD_DIALOG;
 }
 bool main_dialog::on_init_dialog() {
+	set_text(IDC_EDIT1, txt);
 	return true;
 }
 bool main_dialog::on_ok() {
+	txt = get_text(IDC_EDIT1);
 	return true;
 }
 
 void main_window::on_paint(HDC hdc) {
+	RECT r;
+	GetClientRect(*this, &r);
+	HFONT hf = CreateFontIndirect(&lf);
+	SelectObject(hdc, hf);
+
+	if (!str.size())
+		return;
+
+	int x = r.right / 9;
+	int y = r.bottom / str.size();
+	SetTextColor(hdc, color);
+
+	HBRUSH hb = (HBRUSH)GetStockObject(BLACK_BRUSH);
+	for (int i = 0; i < str.length(); i++) {
+		for (int k = 0; k < 8; k++) {
+			const RECT r = { x * k, y * i, x * (k + 1), y * (i + 1) };
+			if ((str[i] & (1 << (7 - k))) == 0) {
+				FillRect(hdc, &r, hb);
+			}
+		}
+		RECT r = { 8 * x, i * y, 9 * x, (i + 1) * y };
+		::DrawText(hdc, &str[i], 1, &r, DT_CENTER | DT_VCENTER | DT_SINGLELINE);
+	}
+	DeleteObject(SelectObject(hdc, hf));

```

Comment:
Nije dobro, ovo neće obrisati `hf` - upravo ste ga još jednom selektirali u DC. Morate selektirati prethodni font (ono što vrati `SelectObject` u liniji 20).